tests/test.rs

```diff
@@ -265,6 +271,38 @@ fn get_compiler_major(version: &String) -> String {
     }
 }
 
+fn create_zip(zip_path: &Path, base_dir: &Path, files_pat: &str) {
+    let mut glob_builder = GlobSetBuilder::new();
+    glob_builder.add(Glob::new(files_pat).unwrap());
+    let globset = glob_builder.build().unwrap();
+    let mut files: Vec<PathBuf> = Vec::new();
+    for entry in WalkDir::new(base_dir) {
+        let entry = entry.expect("Failed to open directory.");
+        if globset.is_match(&entry.file_name()) {
+            files.push(entry.path().to_path_buf());
+        }
+    }
+
+    let zipfile = File::create(base_dir.join(zip_path))
+        .expect(&format!("Cannot create file {:?}", zip_path));
+    let mut zip = ZipWriter::new(zipfile);
+    for ref file_path in files {
+        let mut file = File::open(file_path)
+            .expect(&format!("Cannot open file {:?}", file_path));
+        let file_size = file.metadata()
+            .expect(&format!("Cannot get metadata for {:?}", file_path)).len() as usize;
+        let mut content: Vec<u8> = Vec::with_capacity(file_size + 1);
+        file.read_to_end(&mut content)
+            .expect(&format!("Cannot read {:?}", file_path));
+        zip.start_file(file_path.file_name().unwrap().to_str().unwrap(), FileOptions::default())
+            .expect(&format!("Cannot create zip for {:?}", zip_path));
+        zip.write_all(content.as_slice())
+            .expect(&format!("Cannot write {:?}", zip_path));
+    }
+    zip.finish()
+        .expect(&format!("Unable to write zip structure for {:?}", zip_path));
+}
+
 #[test]
 fn test_integration() {
```

Comment:
Let's add another test that only checks the zip stuff for a single directory (`tests/basic`), since testing on all of them is not needed. This way we can keep `test_integration` simpler and probably have a simpler test for gcno/gcda zip.

Reply:
Doing this will also fix the failure on Mac (since with `tests/basic` we don't have the additional two functions).